packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -1343,3 +1352,151 @@ export function listenToEventResponderEventTypes(
+  left: number | void | null,
+  right: number | void | null,
+  top: number | void | null,
+  bottom: number | void | null,
+  rootContainerInstance: Element | Document,
+  parentNamespace: string,
+): Element {
+  const hitSlopElement = createElement(
+    'div',
+    emptyObject,
+    rootContainerInstance,
+    parentNamespace,
+  );
+  const hitSlopElementStyle = ((hitSlopElement: any): HTMLElement).style;
+
+  hitSlopElementStyle.position = 'absolute';
+  if (top != null) {
+    hitSlopElementStyle.top = `-${top}px`;
+  }
+  if (left != null) {
+    hitSlopElementStyle.left = `-${left}px`;
+  }
+  if (right != null) {
+    hitSlopElementStyle.right = `-${right}px`;
+  }
+  if (bottom != null) {
+    hitSlopElementStyle.bottom = `-${bottom}px`;
+  }
+  return hitSlopElement;
+}
+
+export function diffAndUpdateEventTargetHitSlop(
+  left: number | null | void,
+  right: number | null | void,
+  top: number | null | void,
+  bottom: number | null | void,
+  lastProps: Props,
+  hitSlopElement: HTMLElement,
+): void {
+  const hitSlopElementStyle = hitSlopElement.style;
+  if (lastProps.left !== left) {
+    if (left == null) {
+      hitSlopElementStyle.left = '';
+    } else {
+      hitSlopElementStyle.left = `-${left}px`;
+    }
+  }
+  if (lastProps.right !== right) {
+    if (right == null) {
+      hitSlopElementStyle.right = '';
+    } else {
+      hitSlopElementStyle.right = `-${right}px`;
+    }
+  }
+  if (lastProps.top !== top) {
+    if (top == null) {
+      hitSlopElementStyle.top = '';
+    } else {
+      hitSlopElementStyle.top = `-${top}px`;
+    }
+  }
+  if (lastProps.bottom !== bottom) {
+    if (bottom == null) {
+      hitSlopElementStyle.bottom = '';
+    } else {
+      hitSlopElementStyle.bottom = `-${bottom}px`;
+    }
+  }
+}
+
+export function handleEventTouchHitTarget(
+  lastProps: Props,
+  nextProps: Props,
+  rootContainerInstance: Container,
+  internalInstanceHandle: Object,
+  hostContext: HostContext,
+): void {
+  if (enableEventAPI) {
+    // Validates that there is a single element
+    const node = getElementFromTouchHitTarget(internalInstanceHandle);
+    if (node !== null) {
+      let parentNamespace: string;
+      if (__DEV__) {
+        const hostContextDev = ((hostContext: any): HostContextDev);
+        parentNamespace = hostContextDev.namespace;
+        warning(
+          parentNamespace === HTML_NAMESPACE,
+          'An event touch hit target was used in an unsupported DOM namespace. ' +
+            'Ensure the touch hit target is used in a HTML namespace.',
+        );
+      } else {
+        parentNamespace = ((hostContext: any): HostContextProd);
+      }
+
+      const element = ((node: any): HTMLElement);
+      // We update the event target state node to be that of the element.
+      // We can then diff this entry to determine if we need to add the
+      // hit slop element, or change the dimensions of the hit slop.
+      const lastElement = internalInstanceHandle.stateNode;
```

Comment:
I wasn't aware of this constraint. I can re-work it so that `stateNode` is an object ref instead. Then the object ref is passed to this function in host config; would that make more sense?